Automattic/vip-cookie-banner
# [FEATURE] Reduced unused javascript

Optimized `vip-cookie-banner` plugin by removing the `@automattic/privacy-toolset` dependency, recreating the `<CookieBanner />` and `<DoNotSellDialog />` components, and lazy loading the components.

# Description

- Removed `@automattic/privacy-toolset` dependency
- Recreated Automattic's `<CookieBanner />` component
- Recreated Automattic's `<DoNotSellDialog />` component
- Lazy loaded all components consumed in the `vip-cookie-banner` plugin

#### Related Ticket URLs

[PRS-794](https://app.clickup.com/t/866a95yhr)

#### Screenshots

_Before:_
| Mobile Score  | Desktop Score |
| ------------- | ------------- |
| <img width="1536" alt="image" src="https://github.com/wpcomvip/vip-wordpress-com/assets/20046764/46a99a25-460c-42c7-b888-ed49b84a590b">  | <img width="1536" alt="image" src="https://github.com/wpcomvip/vip-wordpress-com/assets/20046764/c25cbfaf-10a9-4d0c-a921-b07ad5390c09"> |

_After:_
| Mobile Score  | Desktop Score |
| ------------- | ------------- |
| <img width="1536" alt="image" src="https://github.com/wpcomvip/vip-wordpress-com/assets/20046764/93fb5056-cece-4d17-8f1a-cf338dbb9858">  | <img width="1536" alt="image" src="https://github.com/wpcomvip/vip-wordpress-com/assets/20046764/4a569770-5b99-4b27-b4e3-f5f2dbee3c2b"> |

_Component Views:_
| Cookie Banner | Customize Cookie Banner | "Do Not Sell" Dialogue |
| ------------- | ------------- | ------------- |
| <img width="1536" alt="image" src="https://github.com/wpcomvip/vip-wordpress-com/assets/20046764/0dea8f4b-59f7-4ef4-a237-9ec92e7ed857"> | <img width="1536" alt="image" src="https://github.com/wpcomvip/vip-wordpress-com/assets/20046764/19fca1fc-7c91-447b-b3b6-2b9e419b3d02"> | <img width="1536" alt="image" src="https://github.com/wpcomvip/vip-wordpress-com/assets/20046764/0a430c3a-13f0-4f32-ac2a-5105ac0fa40c"> |

## Tested on:


## Steps to Test

1. Visit https://wpvip-valet.go-vip.net/
1. Verify cookie banner and "do not sell" dialogue are functioning as intended
1. Verify improved performance scores at: https://pagespeed.web.dev/
